stl/inc/xstring

```diff
@@ -2432,6 +2438,33 @@ public:
         _Proxy._Release();
     }
 
+    explicit basic_string(_String_constructor_concat_tag, const basic_string& _Alsource, const _Elem* const _Left_ptr,
+        const size_type _Left_size, const _Elem* const _Right_ptr, const size_type _Right_size)
+        : _Mypair(
+            _One_then_variadic_args_t(), _Alty_traits::select_on_container_copy_construction(_Alsource._Getal())) {
+        _STL_INTERNAL_CHECK(_Left_size < max_size());
+        _STL_INTERNAL_CHECK(_Right_size < max_size());
+        _STL_INTERNAL_CHECK(max_size() - _Left_size >= _Right_size);

```

Comment:
Style: I would prefer for all three comparisons to be `<=`; flipping the sense of the third one is more work to reason through.

Reply:
Right, I didn't try to make them consistent because I was using < for some and <= for others, but as you point out in the other comment, that was incorrect. I'll make them consistent.